src/plugins/intel_cpu/src/nodes/unique.cpp

```diff
@@ -0,0 +1,399 @@
+    }
+    if (definedOutputs[OCCURRENCES_NUM]) {
+        occurPtr = reinterpret_cast<int*>(getChildEdgesAtPort(OCCURRENCES_NUM)[0]->getMemoryPtr()->GetPtr());
+    }
+    int64_t uniqLen = inputLen;
+
+    if (sorted) {
+        std::memcpy(uniqueData, srcDataPtr, inputLen * sizeof(T));
+        std::sort(uniqueData, uniqueData + inputLen);
+        auto last = std::unique(uniqueData, uniqueData + inputLen);
+        uniqLen = last - uniqueData;
+
+        if (definedOutputs[FIRST_UNIQUE_IDX]) {
+            T* first = uniqueData;
+            for (T* it = first; it < last; it++) {
+                for (int i = 0; i < inputLen; i++) {
+                    if (srcDataPtr[i] == *it) {
+                        *firstPtr++ = i;
+                        first++;
+                        break;
+                    }
+                }
+            }
+        }
+        if (definedOutputs[INPUT_TO_UNIQ_IDX]) {
+            for (int i = 0; i < inputLen; i++) {
+                if (i > 0 && srcDataPtr[i] == srcDataPtr[i - 1]) {
+                    inToOutPtr[i] = inToOutPtr[i - 1];
+                    continue;
+                }
+                for (int j = 0; j < uniqLen; j++) {
+                    if (srcDataPtr[i] == uniqueData[j]) {
+                        inToOutPtr[i] = j;
+                        break;
+                    }
+                }
+            }
+        }
+        if (definedOutputs[OCCURRENCES_NUM]) {
+            std::fill(occurPtr, occurPtr + uniqLen, 0);
+            for (int j = 0; j < uniqLen; j++) {
+                for (int i = 0; i < inputLen; i++) {
+                    if (srcDataPtr[i] == uniqueData[j]) {
+                        occurPtr[j]++;
+                    }
+                }
+            }
+        }
+    } else {
+        uniqueData[0] = srcDataPtr[0];
+        if (definedOutputs[FIRST_UNIQUE_IDX]) {
+            firstPtr[0] = 0;
+        }
+        if (definedOutputs[INPUT_TO_UNIQ_IDX]) {
+            inToOutPtr[0] = 0;
+        }
+        if (definedOutputs[OCCURRENCES_NUM]) {
+            std::fill(occurPtr, occurPtr + inputLen, 1);
+        }
+        uniqLen = 1;
+
+         for (int i = 1; i < inputLen; i++) {
+             bool found = false;
+             int j = 0;
+             for (; j < uniqLen; j++) {
+                 if (uniqueData[j] == srcDataPtr[i]) {
+                     found = true;
+                     break;
+                 }
+             }
+             if (!found) {
+                 uniqueData[uniqLen] = srcDataPtr[i];
+
+                 if (definedOutputs[FIRST_UNIQUE_IDX]) {
+                     firstPtr[uniqLen] = i;
+                 }
+                 if (definedOutputs[INPUT_TO_UNIQ_IDX]) {
+                     inToOutPtr[i] = j;
+                 }
+
+                 uniqLen++;
+             } else {
+                 if (definedOutputs[INPUT_TO_UNIQ_IDX]) {
```

Comment:
minor: the same for both condition, maybe could be done out of this condition.

Reply:
Fixed